test/OpenTelemetry.Tests/Implementation/Trace/Propagation/CompositePropagatorTest.cs

```diff
@@ -0,0 +1,242 @@
+﻿// <copyright file="CompositePropagatorTest.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using OpenTelemetry.Context.Propagation;
+using Xunit;
+
+namespace OpenTelemetry.Tests.Implementation.Trace.Propagation
+{
+    public class CompositePropagatorTest
+    {
+        private const string TraceParent = "traceparent";
+        private static readonly string[] Empty = new string[0];
+        private static readonly Func<IDictionary<string, string>, string, IEnumerable<string>> Getter = (headers, name) =>
+        {
+            count++;
+            if (headers.TryGetValue(name, out var value))
+            {
+                return new[] { value };
+            }
+
+            return Empty;
+        };
+
+        private static readonly Action<IDictionary<string, string>, string, string> Setter = (carrier, name, value) =>
+        {
+            carrier[name] = value;
+        };
+
+        private static int count = 0;
+
+        private readonly ActivityTraceId traceId = ActivityTraceId.CreateRandom();
+        private readonly ActivitySpanId spanId = ActivitySpanId.CreateRandom();
+
+        [Fact]
+        public void CompositePropagator_NullTextFormatList()
+        {
+            Assert.Throws<ArgumentNullException>(() => new CompositePropagator(null));
+        }
+
+        [Fact]
+        public void CompositePropagator_WithTraceContext()
```

Comment:
I'd remove all TraceState or B3 test from compositepropagator tests. They are tested elsewhere. CompositePropagator doesn't care what its composed of - it just calls inject/extract on them, and the tests must just validate that behavior.

Reply:
ok. im removing everything related to B3 and traceContextFormat.